asset-transfer-events/application-gateway-java/src/main/java/App.java

```diff
@@ -0,0 +1,162 @@
+
+		CompletableFuture.runAsync(() -> {
+			eventIter.forEachRemaining(event -> {
+				String payload = prettyJson(event.getPayload());
+				System.out.println("\n<-- Chaincode event received: " + event.getEventName() + " - " + payload);
+			});
+		});
+
+		return eventIter;
+	}
+
+	private String prettyJson(final byte[] json) {
+		return prettyJson(new String(json, StandardCharsets.UTF_8));
+	}
+
+	private String prettyJson(final String json) {
+		JsonElement parsedJson = JsonParser.parseString(json);
+		return gson.toJson(parsedJson);
+	}
+
+	private long createAsset() throws EndorseException, SubmitException, CommitStatusException {
+		System.out.println("\n--> Submit transaction: CreateAsset, " + assetId + " owned by Sam with appraised value 100");
+
+		SubmittedTransaction commit = contract.newProposal("CreateAsset")
+				.addArguments(assetId, "blue", "10", "Sam", "100")
+				.build()
+				.endorse()
+				.submitAsync();
+
+		Status status = commit.getStatus();
+		if (!status.isSuccessful()) {
+			throw new RuntimeException("failed to commit transaction with status code " + status.getCode());
+		}
+
+		System.out.println("\n*** CreateAsset committed successfully");
+
+		return status.getBlockNumber();
+	}
+
+	private void updateAsset() throws EndorseException, SubmitException, CommitStatusException, CommitException {
+		System.out.println("\n--> Submit transaction: UpdateAsset, " + assetId + " update appraised value to 200");
+
+		contract.submitTransaction("UpdateAsset", assetId, "blue", "10", "Sam", "200");
+
+		System.out.println("\n*** UpdateAsset committed successfully");
+	}
+
+	private void transferAsset() throws EndorseException, SubmitException, CommitStatusException, CommitException {
+		System.out.println("\n--> Submit transaction: TransferAsset, " + assetId + " to Mary");
+
+		contract.submitTransaction("TransferAsset", assetId, "Mary");
+
+		System.out.println("\n*** TransferAsset committed successfully");
+	}
+
+	private void deleteAsset() throws EndorseException, SubmitException, CommitStatusException, CommitException {
+		System.out.println("\n--> Submit transaction: DeleteAsset, " + assetId);
+
+		contract.submitTransaction("DeleteAsset", assetId);
+
+		System.out.println("\n*** DeleteAsset committed successfully");
+	}
+
+	private void replayChaincodeEvents(final long startBlock) {
+		System.out.println("\n*** Start chaincode event replay");
+
+		ChaincodeEventsRequest request = network.newChaincodeEventsRequest(chaincodeName)
+				.startBlock(startBlock)
+				.build();
+
+		try (CloseableIterator<ChaincodeEvent> eventIter = request.getEvents()) {
+			while (eventIter.hasNext()) {
+				ChaincodeEvent event = eventIter.next();
+				String payload = prettyJson(event.getPayload());
+				System.out.println("\n<-- Chaincode event replayed: " + event.getEventName() + " - " + payload);
+
+				if (event.getEventName().equals("DeleteAsset")) {
```

Comment:
could I suggest adding a comment about the importance of the `break`
So it's clear this will keep listening for events indefinately